kratos/utilities/pointer_communicator.h

```diff
@@ -96,6 +96,22 @@ class ResultsProxy
         else
             return mNonLocalData[gp];
     }
+    
+    bool Has(GlobalPointer<TPointerDataType>& gp) 
+    {
+        if(gp.GetRank() == mCurrentRank)
+            return true;
+        else
+            return mNonLocalData.find(gp) != mNonLocalData.end();
+    }   
+
+    bool Has(GlobalPointer<TPointerDataType> const& gp) 
```

Comment:
```suggestion
    bool Has(const GlobalPointer<TPointerDataType>& rGlobalPointer)  const
```

Reply:
Ok, so I will change the Get version too